index.js

```diff
@@ -0,0 +1,299 @@
+(function() {
+  function Calculator(domElement) {
+    this.current = domElement,
+    this.display = this.current.querySelector('.calculator-valueDisplay')
+    this.first = 0,
+    this.second = 0,
+    this.operand = '',
+    this.result = '',
+    this.firstValueAdded = false,
+    this.secondValueAdded = false
+  }
+
+  Calculator.prototype.addTheListeners = function() {
+    var calculator = this
+    var currentCalc = this.current
+    currentCalc.addEventListener('click', function(event){
+      if(event.target.matches('.calculator-button:not(#flip):not(#percentage), #decimal')){
+        calculator.displayValue(event.target.innerHTML)
```

Comment:
Reading information from the DOM like this is slow. Instead give each button its own `data-role` value `<button data-role="5" …>` and then do `const buttonRole = event.target.getAttribute('data-role')` and then switch your logic based on the role of the button pressed.